File: Top 100 Codes/06-Advanced Codes related to Arrays/05-Find Largest sum contiguous Subarray/FindLargestsumcontiguousSubarray.py
def find_largest_sum_contiguous_subarray(arr):
    max_so_far = float('-inf')
    max_ending_here = 0

    for num in arr:
        max_ending_here += num
        if max_so_far < max_ending_here:
            max_so_far = max_ending_here

        if max_ending_here < 0:
            max_ending_here = 0

    return max_so_far

# Example usage
arr = [-2, -3, 4, -1, -2, 1, 5, -3]
max_sum = find_largest_sum_contiguous_subarray(arr)
print("Maximum contiguous subarray sum is:", max_sum)

File: Top 100 Codes/06-Advanced Codes related to Arrays/05-Find Largest sum contiguous Subarray/FindLargestsumcontiguousSubarray.cpp
#include <iostream>
#include <vector>
#include <climits>
using namespace std;

// Function to find the largest sum contiguous subarray
int findLargestSumContiguousSubarray(vector<int>& nums) {
    int maxSoFar = INT_MIN;
    int maxEndingHere = 0;

    for (int i = 0; i < nums.size(); i++) {
        maxEndingHere += nums[i];

        if (maxSoFar < maxEndingHere)
            maxSoFar = maxEndingHere;

        if (maxEndingHere < 0)
            maxEndingHere = 0;
    }

    return maxSoFar;
}

int main() {
    vector<int> arr = {-2, -3, 4, -1, -2, 1, 5, -3};

    int maxSum = findLargestSumContiguousSubarray(arr);
    cout << "Maximum contiguous subarray sum is: " << maxSum << endl;

    return 0;
}
